### PR TITLE
completed_approved_vacation_by_admin

### PR DESCRIPTION
**Pull request by Manish-Srans**
Please, Check It Once And Merge It

### CODE REVIEW

File: src/main/java/com/srans/nestserver/controller/NotificationController.java [hunk 1]
```diff
@@ -27,23 +30,30 @@
...
 	@PreAuthorize("permitAll()")
-	public List<Notification> getAllUserNotification(@PathVariable(value = "userId") Long userId) throws NSException {
+	public List<VacationInfo> getAllUserNotification(@RequestParam Long adminId) throws NSException {
 		logger.info("In::getAllNotificationsOfUser::" );
-		List<Object[]> response = notificationRepo.getAllNotification(userId);
+		List<Object[]> response = notificationRepo.getAllNotification(adminId);
 		System.out.println(response.size());
-		List<Notification> l1 =new ArrayList<>();
-		Notification notifications = null;
+		List<VacationInfo> l1 =new ArrayList<>();
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/NotificationController.java:45]: Variable name not looks good, plz use correct names

File: src/main/java/com/srans/nestserver/controller/NotificationController.java [hunk 2]
```diff
@@ -27,23 +30,30 @@
 	@Autowired
 	private NotificationRepository notificationRepo;
 	
+	@Autowired
+	private VacationRepository vacationRepo;
+	
 	
  
 	//Get Notification	
-	@GetMapping(value = "/users/{userId}/notifications")
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/NotificationController.java:33]: Why u renamed notifications to vacation, then what about notifications?

File: src/main/java/com/srans/nestserver/controller/VacateController.java [hunk 3]
```diff
@@ -57,25 +56,31 @@ public Vacation saveVacationDetails(@Valid @RequestBody Vacation vacation) throw
...
+	 * 
+	 * return vacateService.getVacationDetails(tenantId);
+	 * 
+	 * }
+	 */
 	
-	@GetMapping("users/vacate/approved/{id}")
+	@PutMapping("users/vacate")
 	@PreAuthorize("permitAll()")
-	public String vacationApproved(@PathVariable(value="id") Long tenantId)throws NSException{
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/controller/VacateController.java:74]: Method name should be like verb, approveVaction()

File: src/main/java/com/srans/nestserver/service/VacateService.java [hunk 4]
```diff
@@ -56,69 +56,64 @@
...
+
 	@Autowired
 	private HostelRepository hostelRepo;
-	
+
 	@Autowired
 	private TemplateEngine templateEngine;
-	
+
 	Long floorId;
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/service/VacateService.java:67]: Include private

File: src/main/java/com/srans/nestserver/service/VacateService.java [hunk 5]
```diff
@@ -56,69 +56,64 @@
...
 	Long roomId;
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/service/VacateService.java:68]: Include private

File: src/main/java/com/srans/nestserver/service/VacateService.java [hunk 6]
```diff
@@ -129,38 +124,36 @@ public Vacation processVacation(Vacation vacate) throws MailException, Messaging
...
-		Long hostelid=tenantBookRepo.findHostelId(user.getUserId());
-		
-		tenantBookRepo.findAll().forEach(tenantInfo ->{
-			if(tenantInfo.getTenantId()==user.getUserId()) {
-				 floorId=tenantInfo.getFloorId();
-				 roomId=tenantInfo.getRoomId();
-				 roomRent=tenantInfo.getRoomRent();
-				 roomBedId=tenantInfo.getRoomBedId();
-				 
+		Long hostelid = tenantBookRepo.findHostelId(user.getUserId());
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/service/VacateService.java:141]: Make sure you follow java standards camelcase

File: src/main/java/com/srans/nestserver/repository/NotificationRepository.java [hunk 7]
```diff
@@ -15,7 +15,7 @@
 	public Long[] getNotificationId();
 
 	//get all new notification
-	@Query(value = "select n.id, n.message from notification n  inner JOIN notification_user nu on (n.id=nu.notification_id) where nu.view_status='N' and nu.user_id=?1", nativeQuery = true)
+	@Query(value = "select n.id, n.message, n.tenant_id from notification n  inner JOIN notification_user nu on (n.id=nu.notification_id) where nu.view_status='N' and nu.user_id=?1", nativeQuery = true)
 	public List<Object[]> getAllNotification(Long userId);
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/repository/NotificationRepository.java:19]: Why can't u use jpa query and reduce the tediousnees of handling List objects

File: src/main/java/com/srans/nestserver/service/VacateService.java [hunk 8]
```diff
@@ -129,38 +124,36 @@ public Vacation processVacation(Vacation vacate) throws MailException, Messaging
...
+
+		tenantBookRepo.findAll().forEach(tenantInfo -> {
```
Comment (srans-admin) [inline src/main/java/com/srans/nestserver/service/VacateService.java:143]: FindAll is not good, query to the DB with exact requirement, when db size increases u will see worst time complexity

**Review by srans-admin — COMMENTED** [review]
Requested many changes on you PR

File: src/main/java/com/srans/nestserver/controller/NotificationController.java (hunk 2 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/controller/NotificationController.java:33]: I changed it, Sir

File: src/main/java/com/srans/nestserver/controller/VacateController.java (hunk 3 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/controller/VacateController.java:74]: I have chenged it also..

File: src/main/java/com/srans/nestserver/service/VacateService.java (hunk 4 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/service/VacateService.java:67]: Yes sir, I have included...

File: src/main/java/com/srans/nestserver/service/VacateService.java (hunk 5 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/service/VacateService.java:68]: yes sir, I have included...

File: src/main/java/com/srans/nestserver/repository/NotificationRepository.java (hunk 7 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/repository/NotificationRepository.java:19]: Ok sir, I have changed...

File: src/main/java/com/srans/nestserver/service/VacateService.java (hunk 8 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/service/VacateService.java:143]: Ok sir, I'll changed

File: src/main/java/com/srans/nestserver/service/VacateService.java (hunk 6 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/service/VacateService.java:141]: Yes sir, I'll follow

File: src/main/java/com/srans/nestserver/controller/NotificationController.java (hunk 1 above)
Comment (Manish-Srans) [reply src/main/java/com/srans/nestserver/controller/NotificationController.java:45]: I have renamed reference variable

**Comment by Manish-Srans** [comment]
Sir, I have resolved all issues. I am closing this PR and Raising new PR.

**Pull request closed without merging**